app/components/Navbar.tsx

```diff
@@ -14,29 +14,66 @@ import {
 } from "@/components/ui/sheet";
 
 const links = [
-  { path: "/", text: "About" },
+  { path: "/#about", text: "About" },
   { path: "/publications", text: "Publications" },
   { path: "/contact", text: "Contact Us" },
   { path: "/join", text: "Join Our Team" },
 ];
 
-type NavLinkProps = {
+const getActiveRouteBool = (currentRoute: string, href: string) => {
+  return (
+    currentRoute.includes(href.slice(1)) ||
+    (currentRoute === "/" && href === "/#about")
+  );
+};
+
+type HeaderNavLinkProps = {
+  href: string;
+  text: string;
+};
+
+const HeaderNavLink = ({ href, text }: HeaderNavLinkProps) => {
```

Comment:
I'm going to push back a little here with this one... Is the amount of difference between `HeaderNavLink` and `MobileNavLink` enough to justify the reuse of code? I understand the desire to split them into two separate sub-components, but things like `currentRoute`, `isActiveRoute`, and even the base `Link` structure is the same between the two components.

I would advise this instead, even just for the sake of practicing.

Make a generic `NavLink` component. Props are `href`, `text`, `isMobile`, and `onClick?`. `isMobile` is a boolean used within `NavLink` to determine which `className` to use in the `Link` and determines whether or not to use the `onClick` `div`.

With this design we avoid repeating code, while still maintaining the ability to have modular design and functionality. If we thought more than two types of `NavLink`s could exist we would make `isMobile` into `type` and use a switch statement inside the `NavLink` component.

Hopefully that all makes sense!